includes/gateways/paypal/includes/class-wc-gateway-paypal-response.php

```diff
@@ -63,9 +63,12 @@ protected function get_paypal_order( $raw_custom ) {
 	 * @param  string   $note Payment note.
 	 */
 	protected function payment_complete( $order, $txn_id = '', $note = '' ) {
-		$order->add_order_note( $note );
-		$order->payment_complete( $txn_id );
-		WC()->cart->empty_cart();
+		if( ! get_post_meta($order->get_id(),"_paypal_payment_completed",true) ) {
```

Comment:
This doesn't follow WordPress Coding Standards. Also should avoid using `get_post_meta()` or `update_post_meta()` functions, need to use CRUD of the Orders object.

Reply:
I don't know if there is a order->save(); after. That's why I used updated_post_meta.

Reply:
Yes, need to save after., it's missing `$order->save()`.